include/aws/common/json.h

```diff
@@ -0,0 +1,395 @@
+ * on the object/array containing the aws_json_node.
+ * Deleting this array will also delete any aws_json_nodes it contains.
+ * @return A new array aws_json_node
+ */
+AWS_COMMON_API
+struct aws_json_node *aws_json_array_new(void);
+
+/**
+ * Creates a new boolean aws_json_node with the given boolean and returns a pointer to it.
+ *
+ * Note: You will need to free the memory for the aws_json_node using aws_json_delete on the aws_json_node or
+ * on the object/array containing the aws_json_node.
+ * @param boolean The boolean you want to store in the aws_json_node
+ * @return A new boolean aws_json_node
+ */
+AWS_COMMON_API
+struct aws_json_node *aws_json_boolean_new(bool boolean);
+
+/**
+ * Creates a new null aws_json_node and returns a pointer to it.
+ *
+ * Note: You will need to free the memory for the aws_json_node using aws_json_delete on the aws_json_node or
+ * on the object/array containing the aws_json_node.
+ * @return
+ */
+AWS_COMMON_API
+struct aws_json_node *aws_json_null_new(void);
+
+/**
+ * Creates a new object aws_json_node and returns a pointer to it.
+ *
+ * Note: You will need to free the memory for the aws_json_node using aws_json_delete on the aws_json_node or
+ * on the object/array containing the aws_json_node.
+ * Deleting this object will also delete any aws_json_nodes it contains.
+ * @return
+ */
+AWS_COMMON_API
+struct aws_json_node *aws_json_object_new(void);
+// ====================
+
+// ====================
+// Value setters and getters
+
+/**
+ * Sets the string value of a string aws_json_node.
+ * @param node The string aws_json_node.
+ * @param string The new string value.
+ * @return AWS_OP_SUCCESS if the string was successfully set, otherwise AWS_OP_ERR.
+ */
+AWS_COMMON_API
+int aws_json_string_set(struct aws_json_node *node, char *string);
+
+/**
+ * Sets the number value of a number aws_json_node.
+ * @param node The number aws_json_node.
+ * @param number The new number value.
+ * @return AWS_OP_SUCCESS if the number was successfully set, otherwise AWS_OP_ERR.
+ */
+AWS_COMMON_API
+int aws_json_number_set(struct aws_json_node *node, double number);
+
+/**
+ * Sets the boolean value of a boolean aws_json_node.
+ * @param node The boolean aws_json_node.
+ * @param boolean The new boolean value.
+ * @return AWS_OP_SUCCESS if the boolean was successfully set, otherwise AWS_OP_ERR.
+ */
+AWS_COMMON_API
+int aws_json_boolean_set(struct aws_json_node *node, bool boolean);
+
+/**
+ * Gets the string of a string aws_json_node.
+ * @param node The string aws_json_node.
+ * @return The string of the string aws_json_node, otherwise NULL.
+ */
+AWS_COMMON_API
+char *aws_json_string_get(struct aws_json_node *node);
+
+/**
+ * Gets the number of a number aws_json_node.
+ * @param node The number aws_json_node.
+ * @return The number of the number aws_json_node, otherwise NULL.
+ */
+AWS_COMMON_API
+double *aws_json_number_get(struct aws_json_node *node);
+
```

Comment:
I feel like we're missing an API that returns the type of a node.

Reply:
arguable, in the spirit of a minimal API we could just call the getter for the type we expect and bounce when an error-code is returned